### ISSUE TITLE
Let 'unused variable' be a warning

### ISSUE THREAD

**Issue by dagurval** [CashScript/cashscript#187]
When debugging, it would be nice if errors like

   `Unused variable prefix at Line 71, Column 8`

... could be a warnings instead of compilation error.

This allows for more easilly temporarily commenting out code.

**Comment by mr-zwets** [comment]
yeah definitely need a way to still have the compiled output, maybe with a flag for testing.

This would allow for Test-Driven-Development where the smart contract code is written on the fly based on tests.

**Comment by kiok46** [comment]
I wanted to disable the ‘unused variable’ error or at least convert it into a warning. The error is triggered during the semantic analysis in the `SymbolTableTraversal` class so I added:

```js
if (unusedSymbols.length !== 0) {
    unusedSymbols.forEach((symbol) => {
        console.warn(`[Warning] Unused variable: ${symbol.name} in block`);
    });
    // throw new UnusedVariableError(unusedSymbols[0]);
}
```

Removing the unused symbol using it's index from the tree (`node.statements`) resulted in the expected bytecode. (Only in block, not function/constructor params). One test failed as it expected a throw. I guess Jest’s spyOn can be used to expect a console.warn instead.

I’m curious if there’s been any further discussion on this. What’s the preferred approach here? Should unused variables be stripped from the bytecode, or should they remain? I noticed that Solidity keeps them in the bytecode unless the optimizer is enabled but cashscript's optimisations are enabled by default.

**Comment by rkalis** [comment]
We've moved unused variabels to a compiler warning instead of an error + added the `unused` keyword to indicate an intentionally unused variable. This is live in 0.14.0-next.5.
